app.py

```diff
@@ -190,6 +190,27 @@ def launch_from_entity(self, version=None):
 
         self._launch_app(self.context, version=version)
 
+    def _translate_version_tokens(self, raw_string, version):
+        """
+        Returns string with version tokens replaced by their values. Replaces 
+        {version} and {v0}, {v1}, etc. tokens in raw_string with their values. The {v} 
+        tokens are created by splitting the version string by non-numeric chunks which 
+        can be used to specify parts of the version string by their index. For example:
+            {version} = "9.0v4"
+            {v0} = "9"
+            {v1} = "." 
+            {v2} = "0" 
+            {v3} = "v" 
+            {v4} = "4" 
+        :param raw_string: raw string with un-translated tokens 
+        :param version: version string to use for replacement tokens
+        """
+        version_tokens = re.findall(r'\d+|\D+', version)
```

Comment:
I'm a little worried about this being so digit centric... since there are apps that include letters in portions of their version string (especially during beta testing).  Something like "1.0.12a" wouldn't work with this approach.

Maybe it would be better to allow the version string to have explicit grouping?  Something along the lines of making the setting ```"(9.0)(v4)"``` which would translate into:

```
{version} = "9.0v4"
{v0} = "9.0"
{v1} = "v4"
```

?